cmd/kubetoken/main.go

```diff
@@ -32,15 +33,41 @@ var (
 	dumpJson = kingpin.Flag("json", "dump json").Short('j').Bool()
 )
 
+type keyWordsList []string
+
+func (kwl *keyWordsList) Set(value string) error {
+	if len(value) > 0 {
+		*kwl = append(*kwl, value)
+		return nil
+	}
+	return fmt.Errorf("empty string")
+}
+
+func (kwl *keyWordsList) String() string {
+	return ""
+}
+
+func (kwl *keyWordsList) IsCumulative() bool {
+	return true
+}
+
+// KeyWordsList to convert args into a *[]string
+func KeyWordsList(s kingpin.Settings) *[]string {
+	keyWords := new([]string)
+	s.SetValue((*keyWordsList)(keyWords))
+	return keyWords
+}
+
 func main() {
 	var (
-		user       = kingpin.Flag("user", "StaffID username.").Short('u').Default(os.Getenv("USER")).String()
-		kubeconfig = kingpin.Flag("kubeconfig", "kubeconfig location.").Default(filepath.Join(os.Getenv("HOME"), ".kube", "config")).String()
-		version    = kingpin.Flag("version", "print version string and exit.").Bool()
-		filter     = kingpin.Flag("filter", "only show matching roles.").Short('f').String()
-		namespace  = kingpin.Flag("namespace", "override namespace.").Short('n').String()
-		host       = kingpin.Flag("host", "kubetokend hostname.").Short('h').Default(kubetokend).String()
-		pass       = kingpin.Flag("password", "password.").Short('P').Default(os.Getenv("KUBETOKEN_PW")).String()
+		user         = kingpin.Flag("user", "StaffID username.").Short('u').Default(os.Getenv("USER")).String()
+		kubeconfig   = kingpin.Flag("kubeconfig", "kubeconfig location.").Default(filepath.Join(os.Getenv("HOME"), ".kube", "config")).String()
+		version      = kingpin.Flag("version", "print version string and exit.").Bool()
+		filter       = kingpin.Flag("filter", "only show matching roles.").Short('f').String()
+		namespace    = kingpin.Flag("namespace", "override namespace.").Short('n').String()
+		host         = kingpin.Flag("host", "kubetokend hostname.").Short('h').Default(kubetokend).String()
+		pass         = kingpin.Flag("password", "password.").Short('P').Default(os.Getenv("KUBETOKEN_PW")).String()
+		keyWordsList = KeyWordsList(kingpin.Arg("keywords", "key words list used to filter roles"))
```

Comment:
Add some documentation to the `--filter` and `[<keywords>]` to clarify that `--filter` uses a regex, whereas `[<keywords>]` uses a basic string contains check?

Might be good to clarify as well that the both the filter and keywords have to pass for the role to be included, as well as the role must contain all keywords to pass.

Reply:
addressed in the last commit.